lib/dataset/nanmean.cpp

```diff
@@ -32,4 +39,25 @@ Dataset nanmean(const Dataset &d) {
   return apply_to_items(d, [](auto &&... _) { return nanmean(_...); });
 }
 
+namespace {
+Variable bin_sizes_without_mask_and_nan(const Variable &data) {
+  const auto finite = isfinite(bins_view<DataArray>(data).data());
+  if (const auto mask_union =
+          variable::variableFactory().irreducible_event_mask(data);
+      mask_union.is_valid()) {
+    const auto binned_mask = variable::make_bins_no_validate(
+        data.bin_indices(), variable::variableFactory().elem_dim(data),
+        ~mask_union);
+    return bins_sum(finite & binned_mask);
+  }
+  return bins_sum(finite);
+}
```

Comment:
Why can we not just use `bins_sum(isfinite(data))`? Is the problem the extra cost of copying event coords?